### PR TITLE
Benchmarks for stark prover

### PR DESCRIPTION
**Pull request by jrchatruc**
## Description

This PR adds benchmarks for the STARK prover. Because we don't yet have a way to specify `AIR`s through files, I moved our test `AIR` to a separate shared module so both tests and benchmarks can use them. Eventually they should all live as files in something like an `examples` directory.

Benchmarks are also deployed to gh-pages under the `/bench` path, so once this is merged they'll be on https://lambdaclass.github.io/lambdaworks/bench

## Type of change

- [x] New feature

### CODE REVIEW

File: math/benches/benchmarks/fft.rs [hunk 1]
```diff
@@ -21,7 +21,7 @@ pub fn fft_benchmarks(c: &mut Criterion) {
     let mut group = c.benchmark_group("fft");
     group.sample_size(10); // it becomes too slow with the default of 100
 
-    for pow in 20..=24 {
+    for pow in 20..22 {
```
Comment (jrchatruc) [inline math/benches/benchmarks/fft.rs:24]: This was lowered because the benchmarks take way too long

File: .github/workflows/benchmarks.yml [hunk 2]
```diff
@@ -0,0 +1,36 @@
+name: benchmarks 
+
+on:
+  push:
+    branches: [ main ]
+
+permissions:
+  # deployments permission to deploy GitHub pages website
```
Comment (mpaulucci) [inline .github/workflows/benchmarks.yml:8]: comment seems unrelated?

**Review by mpaulucci — APPROVED** [review]
